PSKoans/Public/Show-Karma.ps1

```diff
@@ -80,6 +85,38 @@ function Show-Karma {
                 $KoanLocation | Invoke-Item
             }
         }
+        'OpenFile' {
+            try {
+                $Results = Get-Karma @GetParams
+            }
+            catch {
+                $PSCmdlet.ThrowTerminatingError($_)
+            }
+            $FilePath = Get-PSKoan -Topic $Results.CurrentTopic.Name -Scope User | Select-Object -ExpandProperty Path
+            $LineNumber = $Results.CurrentTopic.FailedLineNumber
+            if (($PSVersionTable.PSVersion.Major -le 5) -or ($PSVersionTable.Platform -eq "win32nt")) {
+                $Editor = (Get-ItemProperty -Path 'HKCU:\Software\Microsoft\Windows\CurrentVersion\Explorer\FileExts\.ps1\UserChoice').ProgId
+                $Editor =  [regex]::matches($DefaultProgram,'(?<=\\).+(?=.exe)').value
```

Comment:
We shouldn't be getting the editor from registry. It's fine to set this as a default (probably), but we don't handle defaults here. All we need do in this function is grab the _current_ setting from the module settings with `Get-PSKoanSetting -Name Editor` and use that. 🙂 

We don't need to handle platform or any of those things here. If we want to do something like this, that'd be handled when the module is first imported, and we can figure that out in a separate PR.

Reply:
Now I'm confused 😄 
If I'm not mistaken, Get-PSKoanSetting reads the editor from a config file which has a hardcoded value for the editor, do you want to detect the editor and put it in the config file and then read it from there? Should we put the detecting code in the Set-PSKoanSetting or somewhere else? Do you have a better way of detecting the editor other than reading the registry?

Reply:
None of this should be handled in this command 😄 

If we choose to look for some default file association, that should be determined well before Show-Karma is ever called. 🙂 

All this command needs to do is look for the current setting and go from there.

The only place we'd look for this would be a one-time lookup when the module is first imported and the initial settings are created, I think. There's not much point in having an `Editor` setting that users can change if we aren't respecting it here. 😁

Reply:
I put it in Show-Karma to be able to detect any change in the default editor, if we don't check the current default program every time we want to open a file we may end up opening the file in a non-default editor(user may change the default program in windows for .ps1 extension after importing the module). Either we need to check it in Show-Karma or call a function to get the latest result.  Am I right? 🤔

Reply:
The user may prefer Show-Karma to use a specific editor for any number of reasons. I don't think it makes sense to determine that dynamically. Even on windows, file associations are not necessarily reliable as they can be modified by other applications.

I would prefer one-time detection and then if the user wants to change their editor we should allow them to set it manually and have PSKoans respect that setting.

Reply:
Ok no detection here, registry things removed 😉 
Are we good?